GTG/core/xml.py

```diff
@@ -0,0 +1,316 @@
+        task.set_added_date(added)
+    except AttributeError:
+        pass
+
+
+    # Task Tags
+    tags = element.get('tags')
+    tags = [t for t in tags.split(',') if t.strip() != '']
+
+    [task.tag_added(t) for t in tags]
+
+    try:
+        content = element.find('content').text
+
+        # Content includes serialized xml, so it needs to be re-serialized
+        # and then deserialized again.
+        if content:
+            content = f"<content>{content}</content>"
+            content = etree.fromstring(content)
+            etree.strip_tags(content, 'content')
+            content = etree.tostring(content)
+
+            task.set_text(content.decode())
+
+    except AttributeError:
+        # Some tasks might not have a content tag
+        pass
+
+    # Subtasks
+    [task.add_child(sub.text) for sub in element.findall('subtask')]
+
+    return task
+
+
+def task_to_element(task) -> etree.Element:
+    """Serialize task into XML Element."""
+
+    element = etree.Element('task')
+
+    element.set('id', task.get_id())
+    element.set('status', task.get_status())
+    element.set('uuid', task.get_uuid())
+
+    tags = [saxutils.escape(str(t)) for t in task.get_tags_name()]
+    element.set('tags', ','.join(tags))
+
+    title = etree.SubElement(element, 'title')
+    title.text = task.get_title()
+
+    added_date = etree.SubElement(element, 'addeddate')
+    added_date.text = task.get_added_date_string()
+
+    due_date = etree.SubElement(element, 'duedate')
+    due_date.text = task.get_due_date().xml_str()
+
+    modified_date = etree.SubElement(element, 'modified')
+    modified_date.text = task.get_modified_string()
+
+    start_date = etree.SubElement(element, 'startdate')
+    start_date.text = task.get_start_date().xml_str()
+
+    done_date = etree.SubElement(element, 'donedate')
+    done_date.text = task.get_closed_date().xml_str()
+
+    for subtask_id in task.get_children():
+        sub = etree.SubElement(element, 'subtask')
+        sub.text = subtask_id
+
+    text = task.get_text()
+
+    if text:
+        # We take the xml text and convert it to a string
+        # but without the "<content />"
+        text = etree.fromstring(text)
+
+        content = etree.SubElement(element, 'content')
+        content.text = etree.tostring(text)
+
+    return element
+
+
+def get_file_mtime(filepath: str) -> str:
+    """Get date from file."""
+
+    timestamp = os.path.getmtime(filepath)
+    return datetime.fromtimestamp(timestamp).strftime('%Y-%m-%d')
+
+
+def get_backup_name(filepath: str, i: int) -> str:
+    """Get name of backups which are backup/ directory."""
+
+    dirname, filename = os.path.split(filepath)
+    backup_file = f"{filename}.bak.{i}"
+
+    return os.path.join(dirname, 'backup', backup_file)
+
+
+def get_xml_tree(filepath: str) -> etree.Element:
+    """Parse XML file at filepath and get tree."""
+
+    with open(filepath, 'rb') as stream:
+        tree = etree.parse(stream)
```

Comment:
We do use `getroot()` though (`datastore.py`, line 221). Maybe we can use Element once we have the new format, since we will have to redo most of the code anyways